Code/Tools/ProjectManager/Source/CreateAGemScreen.cpp

```diff
@@ -0,0 +1,511 @@
+    {
+        QScrollArea* firstScreenScrollArea = new QScrollArea(this);
+        firstScreenScrollArea->setWidgetResizable(true);
+        firstScreenScrollArea->setMinimumWidth(660);
+        firstScreenScrollArea->setObjectName("createAGemRightPane");
+        QFrame* firstScreenFrame = new QFrame();
+
+        QWidget* firstScreenScrollWidget = new QWidget();
+        firstScreenScrollArea->setWidget(firstScreenScrollWidget);
+
+        QVBoxLayout* firstScreen = new QVBoxLayout(this);
+        firstScreen->setAlignment(Qt::AlignTop);
+        firstScreen->addWidget(firstScreenFrame);
+        firstScreenScrollWidget->setLayout(firstScreen);
+        QLabel* rightPaneHeader = new QLabel();
+        rightPaneHeader->setObjectName("rightPaneHeader");
+        rightPaneHeader->setText(tr("Please Choose a Gem Template"));
+        QLabel* rightPaneSubheader = new QLabel();
+        rightPaneSubheader->setObjectName("rightPaneSubheader");
+        rightPaneSubheader->setText(tr("Gems can contain assets such as scripts, animations, meshes, textures, and more."));
+        firstScreen->addWidget(rightPaneHeader);
+        firstScreen->addWidget(rightPaneSubheader);
+
+        LoadButtonsFromGemTemplatePaths(firstScreen);
+        m_formFolderRadioButton = new QRadioButton("Choose existing template");
+        m_formFolderRadioButton->setObjectName("createAGem");
+        m_radioButtonGroup->addButton(m_formFolderRadioButton);
+
+        m_gemTemplateLocation = new FormFolderBrowseEditWidget(tr("Gem Template Location*"), "");
+
+        firstScreen->addWidget(m_formFolderRadioButton);
+        firstScreen->addWidget(m_gemTemplateLocation);
+
+        return firstScreenScrollArea;
+    }
+
+    QScrollArea* CreateAGemScreen::CreateSecondScreen()
+    {
+        QScrollArea* secondScreenScrollArea = new QScrollArea(this);
+        secondScreenScrollArea->setWidgetResizable(true);
+        secondScreenScrollArea->setMinimumWidth(660);
+        secondScreenScrollArea->setObjectName("createAGemRightPane");
+        QFrame* secondScreenFrame = new QFrame();
+
+        QWidget* secondScreenScrollWidget = new QWidget();
+        secondScreenScrollArea->setWidget(secondScreenScrollWidget);
+
+        QVBoxLayout* secondScreen = new QVBoxLayout(this);
+        secondScreen->setAlignment(Qt::AlignTop);
+        secondScreen->addWidget(secondScreenFrame);
+        secondScreenScrollWidget->setLayout(secondScreen);
+
+        QLabel* secondRightPaneHeader = new QLabel();
+        secondRightPaneHeader->setObjectName("rightPaneHeader");
+        secondRightPaneHeader->setText(tr("Enter Gem Details"));
+        secondScreen->addWidget(secondRightPaneHeader);
+
+        m_gemDisplayName = new FormLineEditWidget(tr("Gem Display name*"), "");
+        QLineEdit* gemDisplayLineEdit = m_gemDisplayName->lineEdit();
+        gemDisplayLineEdit->setPlaceholderText(tr("The name displayed on the Gem Catalog"));
+        secondScreen->addWidget(m_gemDisplayName);
```

Comment:
We should validate on text field changes on text changed so the user doesn't have to keep pushing the next button to see if there is an error
```suggestion
        m_gemDisplayName = new FormLineEditWidget(tr("Gem Display name*"));
        m_gemDisplayName->lineEdit()->setPlaceholderText(tr("The name displayed in the Gem Catalog"));
        connect(m_gemDisplayName->lineEdit(), &QLineEdit::textChanged, [this]{ ValidateGemDisplayName(); });
        secondScreen->addWidget(m_gemDisplayName);
```

If this is a common enough pattern, perhaps we should add a new constructor for `FormLineEditWidget` that accepts the labelText, valueText, placeholderText, and textChanged lambda.  Then we might be able to shorten this to something like

```suggestion
        m_gemDisplayName = new FormLineEditWidget(tr("Gem Display name*"), "", tr("The name displayed in the Gem Catalog"), [this]{ ValidateGemDisplayName(); });
        secondScreen->addWidget(m_gemDisplayName);
```

Reply:
Also, we may not need a custom validator per field if they're just checking simple things like not empty or URL.  We can use existing QValidators like so
```
    m_gemDisplayName ->lineEdit()->SetValidator(new QRegularExpressionValidator(QRegularExpression("\w"), this));
    m_gemName ->lineEdit()->SetValidator(new QRegularExpressionValidator(QRegularExpression("[a-zA-Z]+[a-zA-Z0-9\-\_]?"), this));
```

For the URL stuff you can try converting to the input string to QUrl